wordvecspace/annoy.py

```diff
@@ -85,23 +129,234 @@ def get_distances(self, row_words_or_indices: Union[int, str, np.ndarray],\
+            return self._get_set_intersect(v_w_i, k, examine_k, metric, weights, include_distances)
+        elif combination_method == 'set_union':
+            return self._get_set_union(v_w_i, k, examine_k, metric, weights, include_distances)
+        elif combination_method == 'vector':
+            return self._get_resultant_vector_nearest(v_w_i, k, weights, metric, include_distances)
+        else:
+            raise InvalidCombination()
+
+    def _get_resultant_vector_nearest(self, v_w_i, k, weights, metric, include_distances):
+        """Retrieves nearest indices based on the resultant vector.
+
+        Algorithm:
+        i) Compute the resultant vector by summing per dimension across vectors
+        ii) Commpute distances based on this resultant vector
+        iii) Return nearest indices after sorting
+
+        Tests:
+        >>> wv = WordVecSpaceAnnoy(DATAFILE_ENV_VAR)
+        >>> vecs = wv.get_vectors(['laptop', 'phone'])
+        >>> wv.get_nearest(vecs, k=5, combination=True, combination_method='vector',
+        ... include_distances=True)
+        (array([10101, 27639, 25905, 28303,  8971], dtype=uint32), array([0.0529, 0.0622, 0.1041, 0.1386, 0.1539], dtype=float32))
+        >>> wv.get_nearest(vecs, k=5, combination=True, combination_method='vector',
+        ... include_distances=True)
+        (array([10101, 27639, 25905, 28303,  8971], dtype=uint32), array([0.0529, 0.0622, 0.1041, 0.1386, 0.1539], dtype=float32))
+        """
+
+        v = self._check_vec(v_w_i)
+
+        res = None
+        if not weights:
+            weights = np.ones(len(v_w_i))
+        else:
+            weights = np.array(weights)
 
-        index = self._check_index_or_word(v_w_i)
+        if isinstance(v, (list, np.ndarray)):
+            resultant_vec = (v * weights[:, None]).sum(axis=0, dtype=np.float32)
+            res, distances = self.ann.get_nns_by_vector(resultant_vec, k, include_distances=True)
 
-        return self.ann.get_nns_by_item(index, k)
+        if include_distances:
+            return (np.array(res, dtype=np.uint32), np.array(distances, dtype=np.float32))
+        else:
+            return np.array(res, dtype=np.uint32)
+
+    def _get_nearest(self, v_w_i, k, include_distances):
+        """Retrieves nearest indices when combination=False.
+
+        Tests:
+        >>> wv = WordVecSpaceAnnoy(DATAFILE_ENV_VAR)
+        >>> vecs = wv.get_vectors(['laptop', 'phone'])
+        >>> wv.get_nearest(vecs, k=5)
+        array([[15890, 32156, 31197, 32342, 20573],
+               [ 3826, 10458,  2686, 20515, 39195]], dtype=uint32)
+        """
+
+        if not isinstance(v_w_i, (tuple, list, np.ndarray)):
+            index = self._check_index_or_word(v_w_i)
+
+            nearest_indices, distances = self.ann.get_nns_by_item(index, k, include_distances=True)
+            if include_distances:
+                return (np.array(nearest_indices, dtype=np.uint32), np.array(distances, dtype=np.float32))
+            else:
```

Comment:
even here we do not need an else here and also we don't need next else condition also if a condition returns from there itself we can reduce indents in our code which would be much readable